src/simplewallet/simplewallet.cpp

```diff
@@ -4611,9 +4610,49 @@ bool simple_wallet::transfer_main(int transfer_type, const std::vector<std::stri
   size_t num_subaddresses = 0;
   for (size_t i = 0; i < local_args.size(); i += 2)
   {
-    cryptonote::address_parse_info info;
     cryptonote::tx_destination_entry de;
-    if (!cryptonote::get_account_address_from_str_or_url(info, m_wallet->nettype(), local_args[i], oa_prompter))
+    cryptonote::address_parse_info info;
+    bool r = true;
+
+    // check for a URI
+    std::string address_uri, payment_id_uri, tx_description, recipient_name, error;
+    std::vector<std::string> unknown_parameters;
+    uint64_t amount = 0;
+    bool has_uri = m_wallet->parse_uri(local_args[i], address_uri, payment_id_uri, amount, tx_description, recipient_name, unknown_parameters, error);
+    if (has_uri)
+    {
+      r = cryptonote::get_account_address_from_str_or_url(info, m_wallet->nettype(), address_uri, oa_prompter);
+      if (!payment_id_uri.empty())
+      {
+        if (!tools::wallet2::parse_short_payment_id(payment_id_uri, info.payment_id))
```

Comment:
According to `wallet2::parse_uri`, both short and long payment ID formats seem to be supported. Why limit to short pID only?

Reply:
Because an address cannot contain a long payment id.

Reply:
Come to think of it, the long payment ID is being deprecated, so this way seems fine to me. I still think the long payment ID is allowed in a URI though, at least that's what I see in the unit test `TEST(uri, long_payment_id)`.

Reply:
Indeed it can. This was extracting from the address, where it can't. But I should check for a long here anyway, and not just at the end, you're right.